build/build-plugins.js

```diff
@@ -23,158 +25,113 @@ const plugins = [
-        external: [
-          bsPlugins.Data,
-          bsPlugins.Base,
-          bsPlugins.EventHandler,
-          bsPlugins.Manipulator
-        ],
-        globals: {
-          [bsPlugins.Data]: 'Data',
-          [bsPlugins.Base]: 'Base',
-          [bsPlugins.EventHandler]: 'EventHandler',
-          [bsPlugins.Manipulator]: 'Manipulator'
-        }
-      }
+const mainPlugins = {
+  BaseComponent: 'base-component.js',
+  Alert: 'alert.js',
+  Button: 'button.js',
+  Carousel: 'carousel.js',
+  Collapse: 'collapse.js',
+  Dropdown: 'dropdown.js',
+  Modal: 'modal.js',
+  Offcanvas: 'offcanvas.js',
+  Popover: 'popover.js',
+  ScrollSpy: 'scrollspy.js',
+  Tab: 'tab.js',
+  Toast: 'toast.js',
+  Tooltip: 'tooltip.js'
+}
 
-    default:
-      return {
-        external: []
-      }
-  }
+const nodePlugins = {
+  Popper: '@popperjs/core'
 }
 
-const utilObjects = new Set([
-  'Util',
-  'Sanitizer',
-  'Backdrop'
-])
+const bsPlugins = {
+  ...domPlugins,
+  ...utilPlugins,
+  ...mainPlugins
+}
+const allPlugins = {
+  ...bsPlugins,
+  ...nodePlugins
+}
 
-const domObjects = new Set([
-  'Data',
-  'EventHandler',
-  'Manipulator',
-  'SelectorEngine'
-])
+const makePluginPath = (pluginKey, basePath) => {
+  if (Object.prototype.hasOwnProperty.call(nodePlugins, pluginKey)) {
+    return nodePlugins[pluginKey]
+  }
 
-const build = async plugin => {
-  console.log(`Building ${plugin} plugin...`)
+  const pluginFilename = allPlugins[pluginKey]
 
-  const { external, globals } = getConfigByPluginKey(plugin)
-  const pluginFilename = path.basename(bsPlugins[plugin])
-  let pluginPath = rootPath
+  return path.resolve(__dirname, `${basePath}/${pluginFilename}`)
+}
 
-  if (utilObjects.has(plugin)) {
-    pluginPath = `${rootPath}/util/`
-  }
+const resolvedDist = new Map()
+const resolvedSrc = new Map()
 
-  if (domObjects.has(plugin)) {
-    pluginPath = `${rootPath}/dom/`
-  }
+Object.keys(allPlugins).forEach(key => {
+  resolvedDist.set(key, makePluginPath(key, distPath))
+  resolvedSrc.set(key, makePluginPath(key, srcPath).replace('.js', ''))
+})
+const build = async pluginKey => {
+  console.log(`Building ${pluginKey} plugin...`)
 
+  const pluginFilename = path.basename(bsPlugins[pluginKey])
+  const external = []
+  const globals = {}
   const bundle = await rollup.rollup({
-    input: bsPlugins[plugin],
+    input: resolvedSrc.get(pluginKey),
     plugins,
-    external
+    external: source => {
+      const pattern = /^(\.+)\// // replace starting with ./ or ../
+      // eslint-disable-next-line no-unused-vars
+      const plugin = Object.entries(allPlugins).find(([key, path]) => {
+        return path.includes(source.replace(pattern, ''))
+      })
+
+      if (!plugin) {
+        console.warn(`Source ${source} is not mapped`)
+        return false
+      }
+
+      const pluginKey = plugin[0]
+      external.push(resolvedDist.get(pluginKey))
+      globals[resolvedSrc.get(pluginKey)] = pluginKey
+      return external
```

Comment:
According to https://rollupjs.org/guide/en/#external this should return a boolean (`(id: string, parentId: string, isResolved: boolean) => boolean`)